app/api/delete-event/route.ts

```diff
@@ -0,0 +1,50 @@
+import { getAppServerSession, isAdmin } from "@/app/auth"; // ensuring that only trusted users can access
+import { NextResponse } from "next/server"; // structuring response to users
+import { prisma } from "../db"; // bridge between me and db
+
+/*
+    1. authenticate user: allowDelete
+    2. delete the event from db: deleteEvent
+    3. reflect change to user interface: DELETE
+*/
+
+async function allowDelete(eventId: number, userEmail: string): Promise<boolean> {
+    if (isAdmin(userEmail)) return true;
+    const event = await prisma.event.findUnique({
+        where: { id: eventId },
+        select: { fromEmail: { select: { senderEmail: true } } }
+    });
+    if (!event) return false;
+    return event.fromEmail?.senderEmail === userEmail;
+}
+
+async function deleteEvent(eventId: number): Promise<void> {
+    try {
+        await prisma.event.delete({
+            where: { id: eventId }
+        });
+        console.log(`Successfully deleted event ${eventId}.`)
+    } catch (error) {
+        console.error(`Error deleting the event ${eventId}: `, error); // tells the user what the error is
```

Comment:
(Just to correct any misunderstanding you might have), I think `console.error` here will only prints out to the `npm run dev` console (i.e. the developer), because backend code only prints out to the terminal where frontend code prints out to the browser console.

Reply:
oh you're right- thx for pointing it out. I guess since it's obvious that the logs in route.ts are for developers, I will remove these comments instead of changing my comment.

Reply:
I think it's fine to leave the logging for developers, because if things do go wrong for some reason, it will be easier to know what the error was from looking at the logs without having to reproduce the issue.